src/getCustomLibp2p.ts

```diff
@@ -1,29 +1,31 @@
 import { noise } from '@chainsafe/libp2p-noise'
 import { yamux } from '@chainsafe/libp2p-yamux'
 import { createDelegatedRoutingV1HttpApiClient } from '@helia/delegated-routing-v1-http-api-client'
-import { autoNAT as autoNATService } from '@libp2p/autonat'
 import { bootstrap } from '@libp2p/bootstrap'
 import { circuitRelayTransport } from '@libp2p/circuit-relay-v2'
-import { dcutr as dcutrService } from '@libp2p/dcutr'
-import { identify as identifyService, type Identify } from '@libp2p/identify'
-import { type KadDHT, kadDHT } from '@libp2p/kad-dht'
+import { identify, type Identify } from '@libp2p/identify'
+import { type KadDHT, kadDHT, removePrivateAddressesMapper } from '@libp2p/kad-dht'
 import { mplex } from '@libp2p/mplex'
-import { ping as pingService, type PingService } from '@libp2p/ping'
+import { ping } from '@libp2p/ping'
+import { prometheusMetrics } from '@libp2p/prometheus-metrics'
 import { tcp } from '@libp2p/tcp'
+import { tls } from '@libp2p/tls'
+import { webRTC, webRTCDirect } from '@libp2p/webrtc'
 import { webSockets } from '@libp2p/websockets'
 import { ipnsSelector } from 'ipns/selector'
 import { ipnsValidator } from 'ipns/validator'
-import { createLibp2p as create, type Libp2pOptions } from 'libp2p'
-import { DELEGATED_ROUTING_V1_HOST, USE_DELEGATED_ROUTING } from './constants.js'
+import { createLibp2p as create, type Libp2pOptions, type ServiceFactoryMap } from 'libp2p'
+import isPrivate from 'private-ip'
+import { DELEGATED_ROUTING_V1_HOST, METRICS, USE_DELEGATED_ROUTING } from './constants.js'
 import type { Libp2p, ServiceMap } from '@libp2p/interface'
+import type { PingService } from '@libp2p/ping'
+import type { Multiaddr } from '@multiformats/multiaddr'
 import type { HeliaInit } from 'helia'
 
-interface HeliaGatewayLibp2pServices extends Record<string, unknown> {
+interface HeliaGatewayLibp2pServices extends ServiceMap {
   dht: KadDHT
   delegatedRouting: unknown
   identify: Identify
-  autoNAT: unknown
-  dcutr: unknown
   ping: PingService
```

Comment:
I removed autonat and dcutr because they are only relevant for being able to receive incoming dials, which I don't think we care about here.

Is the ping service required to be a gateway or can that go too?

Reply:
I can't remember off the top of my head if ping was needed, but we're not even exposing that from fastify, so i think we could remove it.. 

I searched tiros and didn't see anything that looks like it's calling ping: https://github.com/search?q=repo%3Aprobe-lab%2Ftiros%20ping&type=code

Reply:
Ok, I've removed it.